runtime/src/qio/sys.c

```diff
@@ -1862,45 +1867,61 @@ err_t sys_socketpair(int domain, int type, int protocol, fd_t* sockfd_out_a, fd_
 extern void chpl_task_yield(void);
 
 err_t sys_select(int nfds, fd_set* readfds, fd_set* writefds, fd_set* exceptfds, struct timeval* timeout, int* nset) {
-
   int got_nset;
   err_t err_out = 0;
-  struct timeval deadline;
-  struct timeval now;
-  struct timeval real_timeout;
-  long wait_usec = 5;
-
-  if (timeout != NULL) {
-    gettimeofday(&deadline, NULL);
-    deadline.tv_sec += timeout->tv_sec;
-    deadline.tv_usec += timeout->tv_usec;
-    if (deadline.tv_usec > 1000000) {
-      deadline.tv_sec++;
-      deadline.tv_usec -= 1000000;
-    }
-  } else if (timeout->tv_sec == 0) {
-    if (timeout->tv_usec < wait_usec)
-      wait_usec = timeout->tv_usec;
+
+  struct timeval second_timeout;
+  struct timeval *first_timeout = NULL;
+  if(timeout != NULL) {
+    first_timeout = (struct timeval *)qio_malloc(sizeof(struct timeval));
```

Comment:
Why are we allocating first_timeout? Why can't it be a regular stack value like second_timeout? I would imagine you could write `struct timeval first_timeout = *timeout;`

Reply:
Oh, is the concern just that timeout could be NULL? Seems like something that can be handled without allocating.

Reply:
🤔 yeah that was the reason cause It won't be possible to point a normal struct to `NULL`. What could be the other way of doing it?

Reply:
made the changes for it to be a stack value